packages/abstraxion/src/components/AbstraxionSignin/index.tsx

```diff
@@ -1,177 +1,31 @@
-        if (data.grants.length > 0) {
-          return true;
-        } else {
-          const delay = Math.pow(2, retries) * 1000;
-          await new Promise((resolve) => setTimeout(resolve, delay));
-          retries++;
-        }
-      } catch (error) {
-        const delay = Math.pow(2, retries) * 1000;
-        await new Promise((resolve) => setTimeout(resolve, delay));
-        retries++;
-      }
-    }
-
-    console.error("Max retries exceeded, giving up.");
-    return false;
-  }
+  const retryRedirect = async () => {
+    const tempAddress = await abstraxionAuth.getKeypairAddress();
+    abstraxionAuth.redirectToDashboard(tempAddress);
+  };
 
   useEffect(() => {
     async function onStartup() {
       try {
-        setIsConnecting(true);
-        // Check for existing keypair and granter address
-        const existingKeypair = await getTempAccount();
-        const existingGranter = localStorage.getItem(
-          "xion-authz-granter-account",
-        );
-        const searchParams = new URLSearchParams(window.location.search);
-        const granter = existingGranter || searchParams.get("granter");
-
-        // If both exist, we can assume user is either 1. already logged in and grants have been created for the temp key, or 2. been redirected with the granter url param
-        // In either case, we poll for grants and make the appropriate state changes to reflect a "logged in" state
-        if (existingKeypair && granter) {
-          const accounts = await existingKeypair.getAccounts();
-          const address = accounts[0].address;
-          setTempAccountAddress(address);
-
-          const pollSuccess = await pollForGrants(address, granter);
-          if (!pollSuccess) {
-            throw new Error("Error polling for grant");
-          }
-
-          setIsConnecting(false);
-          setIsConnected(true);
-          setShowModal(false);
-          setAbstraxionAccount(existingKeypair);
-          configuregranter(granter);
-
-          // Remove query parameter "granted" and "granter"
-          const currentUrl = new URL(window.location.href);
-          currentUrl.searchParams.delete("granted");
-          currentUrl.searchParams.delete("granter");
-          history.pushState({}, "", currentUrl.href);
-        } else if (!existingKeypair || !granter) {
-          // If there isn't an existing keypair, or there isn't a granter in either localStorage or the url params, we want to start from scratch
-          // Generate new keypair and redirect to dashboard
-          const newKeypairaddress = await generateAndStoreTempAccount(); // remove and replace existing keypair
-          const { dashboardUrl } = await fetchConfig(rpcUrl);
-          openDashboardTab(newKeypairaddress, contracts, dashboardUrl);
+        if (!abstraxionAuth) {
+          throw new Error("abstraxion-core not initialized");
         }
+        setIsConnecting(true);
+        await abstraxionAuth.login();
```

Comment:
What happens if this is called multiple times?

Reply:
Well depends on user env/state, but I definitely see your point. See reply to comment below